src/core/Track.cpp

```diff
@@ -1239,75 +1217,35 @@ void TrackContentObjectView::remove( QVector<TrackContentObjectView *> tcovs )
 
 void TrackContentObjectView::copy( QVector<TrackContentObjectView *> tcovs )
 {
-	// Checks if there are other selected TCOs and if so copy them as well
-	if( tcovs.size() > 1 )
-	{
-		// Write the TCOs to a DataFile for copying
-		DataFile dataFile = createTCODataFiles( tcovs );
+	// Write the TCOs to a DataFile for copying
+	DataFile dataFile = createTCODataFiles( tcovs );
 
-		// Add the TCO type as a key to the final string
-		QString finalString = QString( "tco_%1:%2" ).arg( m_tco->getTrack()->type() ).arg( dataFile.toString() );
-
-		// Copy it to the clipboard
-		QMimeData *tco_content = new QMimeData;
-		tco_content->setData( StringPairDrag::mimeType(), finalString.toUtf8() );
-		QApplication::clipboard()->setMimeData( tco_content, QClipboard::Clipboard );
-	}
-	else
-	{
-		tcovs.at(0)->getTrackContentObject()->copy();
-	}
+	// Copy the TCO type as a key and the TCO data file to the clipboard
+	Clipboard::copyStringPair( QString( "tco_%1" ).arg( m_tco->getTrack()->type() ),
+		dataFile.toString() );
 }
 
 void TrackContentObjectView::cut( QVector<TrackContentObjectView *> tcovs )
 {
-	// Checks if there are other selected TCOs and if so cut them as well
-	if( tcovs.size() > 1 )
-	{
-		// Write the TCOs to a DataFile for copying
-		DataFile dataFile = createTCODataFiles( tcovs );
-
-		// Now that the dataFile is created we can delete the tracks, since we are cutting
-		// TODO: Is it safe to call tcov->remove(); on the current TCOV instance?
-		remove( tcovs );
-
-		// Add the TCO type as a key to the final string
-		QString finalString = QString( "tco_%1:%2" ).arg( m_tco->getTrack()->type() ).arg( dataFile.toString() );
+	// Copy the selected TCOs
+	copy( tcovs );
 
-		// Copy it to the clipboard
-		QMimeData *tco_content = new QMimeData;
-		tco_content->setData( StringPairDrag::mimeType(), finalString.toUtf8() );
-		QApplication::clipboard()->setMimeData( tco_content, QClipboard::Clipboard );
-	}
-	else
-	{
-		tcovs.at(0)->cut();
-	}
+	// Now that the TCOs are copied we can delete them, since we are cutting
+	// TODO: Is it safe to call tcov->remove(); on the current TCOV instance?
```

Comment:
It is safe to call `TrackContentObjectView::remove` on the current TCOV - it results in a call to `QWidget::close`, which is safe to call on the current widget.